core-bundle/src/Resources/contao/library/Contao/Widget.php

```diff
@@ -741,6 +743,23 @@ public function validate()
 		}
 
 		$this->varValue = $varValue;
+
+		$blnFinalize = true;
+
+		if (\func_num_args() > 0)
+		{
+			$blnFinalize = func_get_arg(0);
+		}
+
+		if ($blnFinalize)
+		{
+			@trigger_error('Finalizing a widget during Widget::validate() has been deprecated and will no longer work in Contao 5.0.', E_USER_DEPRECATED);
+
+			if ($this instanceof FinalizableWidgetInterface)
+			{
+				$this->finalize();
```

Comment:
Should we only deprecate `validate()` (without `false`) for finalizable widgets?

```suggestion
			if ($this instanceof FinalizableWidgetInterface)
			{
				@trigger_error('Finalizing a widget during Widget::validate() has been deprecated and will no longer work in Contao 5.0. Use Widget::validate(false) and Widget::finalize() instead.', E_USER_DEPRECATED);

				$this->finalize();
```